server/routes/github/github_oauth.py

```
@@ -0,0 +1,341 @@
+
+    backend_url = (os.getenv("NEXT_PUBLIC_BACKEND_URL", "") or "").rstrip("/")
+    redirect_uri = (
+        f"{backend_url}/github/callback"
+        if backend_url
+        else f"{request.host_url.rstrip('/')}/github/callback"
+    )
+
+    try:
+        signed_state = _sign_state(user_id)
+    except RuntimeError:
+        logger.error("[GITHUB-OAUTH] FLASK_SECRET_KEY missing; cannot sign state")
```

Ruff (0.15.12)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)